app/components/ui/Carousel.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const CarouselOuter = styled.div`
+  position: relative;
+  overflow: visible;
+  width: 100%;
+`;
+
+const CarouselContainer = styled.div`
+  overflow: hidden;
+`;
+
+const CarouselSlider = styled.div`
+  position: relative;
+  display: flex;
+  flex-wrap: nowrap;
+  transition: transform 1s ease;
+  transform: translateX(${props => -props.activeIndex * props.slotWidth}%);
+`;
+
+const CarouselNavButton = ({ dir, onClick }) => (
+  <button type="button" className={`carousel-nav btn-${dir}`} onClick={onClick} />
```

Comment:
Since you've built the rest of the components on this page with Styled Components, maybe it makes sense to translate these classes to Styled Components as well?

Reply:
I'll do that. That question about Styled Components - I was hesitant to introduce them, but I didn't see an easy way to implement a scalable Carousel without the help of dynamically-generated CSS. If there's an idiomatic way to do this in Sass I should read up on, I'd be interested!

Reply:
I'm not opposed to using Styled Components, since I think it makes it more difficult to create overly "global" styles, but in this particular case. I think you may just be able to use (S)CSS for normal styling, and then use inline styles via the `style` prop for anything that must be dynamic.

Also, have you considered using an existing library for creating the carousel and tweaking the styles to match our design?

Reply:
Cool, was able to do that. It seemed like existing libraries that matched what we needed were pretty scarce. I think one that could show multiple items at once also had a couple of extra features that would be pretty expensive (e.g. triggering rerenders via a timer to cycle through the slides), so I figured it wouldn't be too bad to try to just do it on my own. Turns out it took a bit longer to get the carousel to work with the mobile view!